tests/python/pants_test/engine/test_rules.py

```diff
@@ -214,7 +214,8 @@ def a_from_b_noop(b: B) -> A:
 
     self.assert_equal_with_printing(dedent("""
                      Rules with errors: 1
-                       (A, [B], a_from_b_noop()):
+                       (A<a_from_b_noop()>,
```

Comment:
The new `fmt_rule()` util will hopefully make these tests easier to update :) 

I recommend making your first change be to update `test_rules.py` to use `fmt_rule()` - I wish I did this way earlier along in the rule graph errors PR. I wasted so much time trying to manually update every test. (Although, if you rebase, they already will be using this - so, really, it's updating `testutil.engine.util.fmt_rule`)

Also, you'll want to make sure that `test_scheduler.py` and `test_engine.py` are fine when changing anything with rule visualization. Those depend on how we visualize rules.

Reply:
I've since added several new methods alongside `fmt_rule` in `util.py` to format graph nodes. This means we'll only have to change one place the next time we modify how engine graph nodes are output.